File: leetcode/0038-count-and-say/Solution.java
class Solution {
    public String countAndSay(int n) {
        String result = "1";
        for (int k = 0; k < n - 1; k++) {
            StringBuilder next = new StringBuilder();
            int i = 0;
            while (i < result.length()) {
                int j = i;
                while (j < result.length() && result.charAt(j) == result.charAt(i)) {
                    j++;
                }
                next.append(j - i);
                next.append(result.charAt(i));
                i = j;
            }
            result = next.toString();
        }
        return result;
    }
}

File: leetcode/0038-count-and-say/solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    string countAndSay(int n) {
        string result = "1";
        for (int k = 0; k < n - 1; k++) {
            string next;
            int i = 0;
            while (i < (int)result.size()) {
                int j = i;
                while (j < (int)result.size() && result[j] == result[i]) {
                    j++;
                }
                next += to_string(j - i);
                next += result[i];
                i = j;
            }
            result = next;
        }
        return result;
    }
};
